Тестирование работы Controller'a и Request'a для конечной точки регистрации

rotat1ck opened the pull request:
## Controllers
* Добавлен файл `auth_controller.py` содержащий тестируемую функцию `register`

## Requests
* Добавлены файлы:
  ```
  app/requests/auth/login_request.py
  app/requests/auth/register_request.py
  ```
* Файл `register_request.py` проводит валидацию для функции в контроллере
* Валидация проверяется по булевой переменной внутри возвращаемого объекта

eulalourens00 committed:
auth

eulalourens00 committed:
try

rotat1ck committed:
Почти рабочий endpoint /api/v1/auth/register

rotat1ck committed:
AuthController | register() - добавил создание пользователя по переданным параметрам

Evollt commented on app/controllers/auth_controller.py:
```diff
@@ -0,0 +1,29 @@
+from hashlib import sha256
+
+class AuthController(Controller):
+    @staticmethod
```
зачем здесь указывать декоратор staticmethod? Разве в python если метод не принимает self, то он не статичный?

Evollt commented on app/controllers/auth_controller.py:
```diff
@@ -0,0 +1,29 @@
+        db.session.add(new_user)
+        db.session.commit()
+        
+        return jsonify({"message": "User successfully created"}), 200
```
Возвращаем объект пользователя, а не сообщение, что он создан

Evollt commented on app/requests/auth/login_request.py:
Если файл не нужен, то удаляем его. Не надо на dev пихать пустые файлы

Evollt commented on app/requests/auth/register_request.py:
```diff
@@ -0,0 +1,23 @@
+        if params.get("password") is None:
+            not_validated["is_failure"] = True
+            not_validated["errors"].append({"message": "Password is required"})
+        
```
Вот здесь проверь длину массива errors и если она больше единицы, то меня **is_failure** на true, иначе оставь false

Evollt commented on app/requests/auth/register_request.py:
```diff
@@ -0,0 +1,23 @@
+            not_validated["is_failure"] = True
+            not_validated["errors"].append({"message": "Email is required"})
+        #password
+        if params.get("password") is None:
```
Это можно перенести в родительский класс Request.

Evollt commented on app/requests/auth/register_request.py:
```diff
@@ -0,0 +1,23 @@
+            not_validated["is_failure"] = True
+            not_validated["errors"].append({"message": "Username is required"})
+        #email
+        if params.get("email") is None:
```
Перенеси проверку почты в родительский класс Request

rotat1ck committed:
AuthController | register - удалил @staticmethod, RegisterRequest - изменение булевой переменной единожды

rotat1ck committed:
Добавил функцию для сереализации объектов типа UserModel, перенес валидацию имейла и пароля в base класс Request

К слову - `@staticmethod` в функциях для валидации необходим. Оказывается просто убрав self не делает метод статичным. Его нельзя будет вызвать из функций дочерних классов не являющимися статичными.

rotat1ck committed:
Merge branch 'dev' into authorization

Evollt commented on app/requests/auth/register_request.py:
```diff
@@ -0,0 +1,38 @@
+        #email
+        email = params.get("email")
+        
+        if email is None:
```
Эту проверку кидаем в self.email

Evollt commented on app/requests/auth/register_request.py:
```diff
@@ -0,0 +1,38 @@
+        #password
+        password = params.get("password")
+        
+        if password is None:
```
Кидаем в self.password

rotat1ck committed:
Перенос проверки получения параметров password и email в request

rotat1ck commented:
Исправлено

Evollt commented on base/request.py:
```diff
@@ -1,2 +1,48 @@
+        }
+        pattern = r"^.{8,}$"
+        
+        if password is None:
+            not_validated["is_failure"] = True
+            not_validated["error"] = "Password is required"
+            return not_validated
+        
+        if re.match(pattern, password) is None:
+            not_validated["is_failure"] = True
+            not_validated["error"] = "Entered password doesn't satisfy the requirements"
+            return not_validated
+        
+        return not_validated
```
Сделай проверку через if и elif и избавься таким образом от повторения
```python
return not_validated
```

Evollt commented on base/request.py:
```diff
@@ -1,2 +1,48 @@
+        }
+        pattern = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+        
+        if email is None:
+            not_validated["is_failure"] = True
+            not_validated["error"] = "Email is required"
+            return not_validated
+        
+        #check
+        if re.match(pattern, email) is None:
+            not_validated["is_failure"] = True
+            not_validated["error"] = "Entered email doesn't satisfy the requirements"
+            return not_validated
+
+        return not_validated
```
Сделай проверку через if и elif и избавься таким образом от повторения
```python
return not_validated
```

rotat1ck committed:
Изменил кострукцию if'ов, добавил пару комментов

rotat1ck committed:
Изменение логики получения ошибок от request.py

rotat1ck committed:
Я даун соре
